src/routes/customers.php

```diff
@@ -15,8 +15,147 @@
+        }
+    } catch (PDOException $e) {
+        $db->rollBack();
+        $db = null;
+        $response->write($e);
+        return $response->withStatus(500);
+    }
+});
+
+$app->post('/api/customers/signup', function(Request $request, Response $response){
+    $first_name = $request->getParam('first_name');
+    $last_name = $request->getParam('last_name');
+    $username = $request->getParam('username');
+    $password = $request->getParam('password');
+    $email = $request->getParam('email');
+    $phone = $request->getParam('phone');
+    $sql = "INSERT INTO customers(first_name, last_name, username, password, email, phone, points) VALUES (:first_name, :last_name, :username, :password, :email, :phone, 0)";
+    // prepare query
+
+    try{
+        // Get DB Object
+        $db = new db();
+        // Connect
+        $db = $db->connect();
+        $stmt = $db->prepare($sql);
+        $stmt->bindParam(':first_name', $first_name);
+        $stmt->bindParam(':last_name',  $last_name);
+        $stmt->bindParam(':username', $username);
+        $stmt->bindParam(':password',  $password);
+        $stmt->bindParam(':email', $email);
+        $stmt->bindParam(':phone',  $phone);
+        $stmt->execute();
+        $db = null;
+        $response->write('{"notice": {"text": "Customers Added"}}');
+        return $response->withStatus(200);
+    } catch(PDOException $e){
+        $db = null;
+        $response->write('{"error": {"text": '.$e->getMessage().'}}');
+        return $response->withStatus(500);
+    }
+});
+//update by username
+$app->put('/api/customers/update', function (Request $request, Response $response) {
+    $first_name = $request->getParam('first_name');
+    $last_name = $request->getParam('last_name');
+    $username = $request->getParam('username');
+    $password = $request->getParam('password');
+    $email = $request->getParam('email');
+    $phone = $request->getParam('phone');
+    $address_id = $request->getParam('address_id');
+    $points = $request->getParam('points');
+    $sql = "UPDATE customers
+         SET first_name = :first_name, last_name = :last_name,  password = :password, phone = :phone,
+              email = :email, address_id = :address_id, points = :points
+         WHERE username = :username";
+    try {
+        // Get DB Object
+        $db = new db();
+        // Connect
+        $db = $db->connect();
+        $stmt = $db->prepare($sql);
+        $stmt->bindParam(':first_name', $first_name);
+        $stmt->bindParam(':last_name',  $last_name);
+        $stmt->bindParam(':password', $password);
+        $stmt->bindParam(':phone',  $phone);
+        $stmt->bindParam(':email', $email);
+        $stmt->bindParam(':points', $points);
+        $stmt->bindParam(':address_id', $address_id);
+        $stmt->bindParam(':username', $username);
+        $stmt->execute();
+        $db = null;
+        return $response->withStatus(200);
+    } catch (PDOException $e) {
+        $db->rollBack();
```

Comment:
you don't need rollback here, also check hotel.php code for the correct usage of rollback function